doc/design/model_zoo_pai.md

```diff
@@ -0,0 +1,134 @@
+-- Train
+SELECT * FROM train_table ... TO TRAIN MyAwesomeEstimator ... INTO my_awesome_model;
+```
+
+```sql
+-- Predict
+SELECT * FROM pred_table ... TO PREDICT using my_awesome_model ... ;
+```
+
+```sql
+-- Analyze
+SELECT * FROM train_table ...  TO ANALYZE my_awesome_model USING TreeExplainer;
+```
+
+The extended SELECT statements are executed as described in [Model Zoo Design Doc](model_zoo.md#Submitter-Programs):
+- If the model **is not** from a model zoo, The SQLFlow server converts these statements to a *submitter* program and submits the program to a specified engine.
+- If the model **is** from a model zoo, the submmiter program is mounted into a Docker image. The SQLFlow server calls k8s API to launch the Docker image on a k8s cluster.
+
+## Concepts
+
+Besides what described in [Model Zoo Concepts](model_zoo.md#Concepts), there are several new concepts about Alibaba PAI.
+
+1. The **PAI platform** or **PAI** for short is an end-to-end platform that provides various machine learning algorithms to meet your data mining and analysis requirements. PAI supports TensorFlow and many other machine learning frameworks. See [PAI Introduction](https://www.alibabacloud.com/help/doc-detail/67461.htm) for more details.
+
+1. **PAI TensorFlow** is the deployment of TensorFlow on **PAI**, with necessary optimization and further development that makes it possible to cooperate efficiently with other [Alibaba Cloud](https://www.alibabacloud.com/) components such as MaxCompute.
+
+1. A **PAI program** is a python program that is developed base on TensorFlow, MXNet or [other machine learning frameworks supported by PAI](https://www.alibabacloud.com/help/doc-detail/69688.htm).
+
+1. A **PAI task** is an instance of a **PAI program** that is being executed by the **PAI platform**.
+
+## The Design
+
+### Versioning and Releasing
+
+Versioning and releasing in PAI model zoo is the same as what's described in [Model Zoo Design](model_zoo.md#Versioning-and-Releasing). The only requirement is that the base docker image of the SQLFlow model zoo should incorporate both `odpscmd`(which is already in place) and `pyodps`.  We propose to use `odpscmd` at the moment because it requires less code modification in both python and golang.
+
+### Submitter Programs of PAI Model Zoo
+
+Currently, each deployment of SQLFlow has been configured to use only one submitter. So we assume that all the tasks of the deployment of SQLFlow on PAI will be submitted to PAI.
+
+When a user submits a SELECT statement as above, SQLFlow should take the following actions:
+
+1. SQLFlow Checks whether the entity after `TO TRAIN/PREDICT/ANALYZE` is from a SQLFlow model zoo. For example, a plain `DNNClassier` implies that the model is a premade estimator, and `"models.sqlflow.org/sqlflow/my_awesome_model"` implies that the model is from model zoo "models.sqlflow.org". The actual mechanism may be more complicated and is still under progress.
+1. Case A, the model **is not** from a model zoo:
```

Comment:
couler or workflow uses python API to submit a workflow, better to use PAI python API here?

Reply:
Done.